### PR TITLE
Cache key map

### PR DESCRIPTION
**Pull request by naxvog**
Inspired by #270 I have added a cache key map to avoiding consecutive builds of already built redis cache keys.

### CODE REVIEW

File: includes/object-cache.php [hunk 1]
```diff
@@ -284,6 +284,13 @@ class WP_Object_Cache {
      */
     public $cache = [];
 
+    /**
+     * Holds cache key to redis key map
+     *
+     * @var array
+     */
+    public $keys = [];
```
Comment (tillkruss) [inline includes/object-cache.php:292]: Since it's centrally managed, should we use a `static $keys = [];` inside the method itself?

**Comment by tillkruss** [comment]
Great work 👍🏻

**Comment by ocean90** [comment]
Just noting that this change broke the use of the drop-in in Multisite. Applying fbfc86fc22b4df0c636b05e5ce5da0b548fd5b08 and flushing the cache will fix it.

**Comment by naxvog** [comment]
```php
$prefix = $this->is_global_group( $group ) ? $this->global_prefix : $this->blog_prefix;
```
`blog_prefix` contains `blog_id` that's why the error occurs - need to consider `$prefix` in `$key_index`. Will post a fixed version shortly.

**Comment by tillkruss** [comment]
@ocean90: I tagged 2.0.15 👍🏻

**Pull request merged**